src/transaction/txParser.ts

```diff
@@ -43,13 +47,63 @@ const parseTxInputs = (
   return txInputs.map(([txHash, outputIndex]): _Input => ({ txHash, outputIndex }))
 }
 
-const parseTxOutputs = (
-  txOutputs: any[],
-): _Output[] => {
+const parseAsset = (asset: any): _Asset => {
+  if (asset instanceof Map) {
+    return Array.from(asset.entries()).reduce(
+      (acc, [assetName, amount]) => {
+        if (Buffer.isBuffer(assetName)) {
+          try {
+            acc.set(Buffer.from(assetName), BigInt(amount))
+          } catch {
+            throw Error(Errors.TxMultiAssetAmountParseError)
+          }
+        } else throw Error(Errors.TxAssetNameParseError)
+        return acc
+      },
+      new Map<Buffer, BigInt>(),
+    )
+  }
+  throw Error(Errors.TxAssetParseError)
+}
+
+const parseMultiAsset = (multiAsset: any): _MultiAsset => {
+  if (multiAsset instanceof Map) {
+    return Array.from(multiAsset.entries()).reduce(
+      (acc, [policyId, asset]) => {
+        if (Buffer.isBuffer(policyId)) {
+          acc.set(Buffer.from(policyId), parseAsset(asset))
+        } else throw Error(Errors.TxMultiAssetPolicyIdParseError)
+        return acc
+      },
+      new Map<Buffer, Map<Buffer, BigInt>>(),
+    )
+  }
+  throw Error(Errors.TxMultiAssetParseError)
+}
+
+const parseTxOutputs = (txOutputs: any[]): _Output[] => {
   if (!isArrayOfType<TxOutput>(txOutputs, isTxOutput)) {
-    throw Error(Errors.TxOutputParseError)
+    throw Error(Errors.TxOutputParseArrayError)
+  }
+
+  const parseLovelace = (coin: any): Lovelace => {
+    try {
+      return BigInt(coin)
+    } catch {
+      throw Error(Errors.TxOutputParseCoinError)
+    }
+  }
+
+  const parseAmount = (value: any): { coins: BigInt, tokenBundle?: _MultiAsset } => {
+    try {
+      return { coins: parseLovelace(value) }
+    } catch {
+      const [coin, multiAsset] = value
+      return { coins: parseLovelace(coin), tokenBundle: parseMultiAsset(multiAsset) }
+    }
```

Comment:
Im not sure whether we need this try catch logic, would something like this work better? 
```
    if (isLovelace(value)) {
      return { coins: parseLovelace(value) }
    }
    const [coin, multiAsset] = value
    return { coins: parseLovelace(coin), tokenBundle: parseMultiAsset(multiAsset) }
```

Reply:
Refactored